qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/ls/commons/RPCValidatorDelayer.java

```diff
@@ -0,0 +1,89 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.ls.commons;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+/**
+ * Validate a given text document with a delay.
+ *
+ * No model is passed to the validator, since it is assumed that the validation
+ * will be performed by a remote procedure that has access to the model already.
+ *
+ * @see {@ModelValidatorDelayer}
+ */
+public class RPCValidatorDelayer<T extends TextDocument> {
```

Comment:
As there is similar code with`ModelValidatorDelayer`, I think it should be good to have an abstract class `AbstractValidatorDelayer` which manages the delayer feature and `ModelValidatorDelayer` and your class could extend it.